crawl-ref/source/l-moninf.cc

```diff
@@ -321,6 +325,39 @@ static int moninf_get_ev(lua_State *ls)
     return 1;
 }
 
+/*** The string displayed if you target this monster.
+ * Returns the string displayed if you target this monster with a weapon, spell, or item.
+ * @tparam[opt] string|item spell name (string) or item (item object); default=you.weapon()
+ * @treturn string targeting description
+ * @function target_desc
+ */
+static int moninf_get_target_desc(lua_State *ls)
+{
+    MONINF(ls, 1, mi);
+    if (lua_isnone(ls, 2) || lua_isnil(ls, 2))
+    {
+        // default is target description with your current weapon
+        ostringstream result;
+        describe_to_hit(*mi, result, false);
+        lua_pushstring(ls, result.str().c_str());
+        return 1;
+    }
+    if (lua_isuserdata(ls, 2))
+    {
+        // to-hit with a thrown item
+        item_def *item = *(item_def **) luaL_checkudata(ls, 2, ITEM_METATABLE);
+        ranged_attack attk(&you, nullptr, item, is_pproj_active());
```

Comment:
This is a confusing api, I would expect if I put a melee weapon in I would get my melee to-hit and not my chance to hit throwing a fumble.

Reply:
@ebering I've separated this into _four_ functions -- one for the monster itself (there is useful information in the targeter description that is shown nowhere else), and three for the success descriptions given by targeting with a weapon, spell, or thrown item.

Reply:
Four functions! Excellent.

Sorry for the merge delay, your updated version arrived close to the run up to release.